<robot
  name="rover19_urdf">
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <mass
        value="2.7918" />
      <inertia
        ixx="0.095684"
        ixy="0.001902"
        ixz="-0.00027184"
        iyy="0.058185"
        iyz="0.00031126"
        izz="0.14455" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/rover.stl"  scale="0.001 0.001 0.001"/>
      </geometry>
      <material
        name="">
        <color
          rgba="0.84314 0.81569 0.75294 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/rover.stl" />
      </geometry>
    </collision>
  </link>
 
  <link
    name="base_rot">
    <inertial>
      <origin
      xyz="0 0 0 "
      rpy="0 0 0" />
      <mass
        value="11.53" />
      <inertia
        ixx="0.13597"
        ixy="1.2879E-07"
        ixz="-9.0853E-09"
        iyy="0.097658"
        iyz="-3.0537E-14"
        izz="0.097658" />
    </inertial>
    <visual>
      <origin
      xyz="0 0 0"
      rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/base_rot.stl"  scale="0.001 0.001 0.001"/>
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
      xyz="0 0 0"
      rpy="0 0 0"/>
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/base_rot.stl" />
      </geometry>
    </collision>
  </link>
  <joint
    name="base_rot_joint"
    type="revolute">
    <origin
      xyz="0.25289 -0.01547 0.24660"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="base_rot" />
    <axis
      xyz="0 0 1" />
    <limit lower="-1" upper="1" effort="0" velocity="0"/>
  </joint>

  
  <link
    name="shoulder">
    <inertial>
      <origin
      xyz="0 0 0"
      rpy="0 0 0" />
      <mass
        value="11.53" />
      <inertia
        ixx="0.13597"
        ixy="1.2879E-07"
        ixz="-9.0853E-09"
        iyy="0.097658"
        iyz="-3.0537E-14"
        izz="0.097658" />
    </inertial>
    <visual>
      <origin
      xyz="0.12983 -0.00093 0.22673"
      rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/shoulder.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
      xyz="0 0 0"
      rpy="0 0 0"/>
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/shoulder.stl" />
      </geometry>
    </collision>
  </link>
  <joint
    name="shoulder_joint"
    type="continuous">
    <origin
      xyz="0 0 0"
      rpy="0 0 0" />
    <parent
      link="base_rot" />
    <child
      link="shoulder" />
    <axis
      xyz="0 1 0" />
      <limit lower="-1" upper="1" effort="0" velocity="0"/>
  </joint>

  <link
    name="elbow">
    <inertial>
      <origin
      xyz="0 0 0"
      rpy="0 0 0" />
      <mass
        value="11.5298327680895" />
      <inertia
        ixx="0.135971148628003"
        ixy="-5.63856091752789E-08"
        ixz="91.16335845630891E-07"
        iyy="0.0976582051487171"
        iyz="-1.71203329291103E-13"
        izz="0.0976582051489874" />
    </inertial>
    <visual>
      <origin
        xyz="0.17239 0.00432 0.08975"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/elbow.stl" scale="0.001 0.001 0.001" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/elbow.stl" />
      </geometry>
    </collision>
  </link>
  <joint
    name="elbow_joint"
    type="continuous">
    <origin
      xyz="0.23416 -0.00984 0.46051"
      rpy="0 0 0" />
    <parent
      link="shoulder" />
    <child
      link="elbow" />
    <axis
      xyz="0 1 0" />
  </joint>

  <link
    name="gripper_link1">
    <inertial>
      <origin
      xyz="0 0 0"
      rpy="0 0 0" />
      <mass
        value="11.5298327680895" />
      <inertia
        ixx="0.135971148628439"
        ixy="-2.13278075453915E-13"
        ixz="1.49984791894007E-14"
        iyy="0.0976582051486343"
        iyz="2.16840434497101E-18"
        izz="0.0976582051486343" />
    </inertial>
    <visual>
      <origin
        xyz="0.05041 0.00009 -0.02947"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/gripper_link1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/gripper_link1.stl" />
      </geometry>
    </collision>
  </link>
  <joint
    name="gripper_link1_joint"
    type="continuous">
    <origin
      xyz="0.34291 0.00174 0.13836"
      rpy="0 0 0" />
    <parent 
      link="elbow" />
    <child
      link="gripper_link1" />
    <axis
      xyz="0 1 0" />
  </joint>
 
  <link
    name="gripper_link2">
    <inertial>
      <origin 
      xyz="0 0 0"
      rpy="0 0 0" />
      <mass
        value="11.5298327680895" />
      <inertia
        ixx="0.135971148628439"
        ixy="-2.13278075453915E-13"
        ixz="1.49984791894007E-14"
        iyy="0.0976582051486343"
        iyz="2.16840434497101E-18"
        izz="0.0976582051486343" />
    </inertial>
    <visual>
      <origin
        xyz="0.05071 -0.00319 -0.03127"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/gripper_link2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/gripper_link2.stl" />
      </geometry>
    </collision>
  </link>
  <joint
    name="gripper_link2_joint"
    type="revolute">
    <origin
      xyz="0.11327 -0.00419 -0.05624"
      rpy="0 0 0" />
    <parent
      link="gripper_link1" />
    <child
      link="gripper_link2" />
    <axis
      xyz="55.32 -0.68 -36.51" />
    <limit lower="-1" upper="1" effort="0" velocity="0"/>
  </joint>

  <link
    name="finger0">
    <inertial>
      <origin
      xyz="0 0 0"
      rpy="0 0 0" />
      <mass
        value="11.5298327680895" />
      <inertia
        ixx="0.135971148628439"
        ixy="-1.95838647067939E-13"
        ixz="-9.49762849544681E-14"
        iyy="0.097658205148634"
        iyz="2.42861286636753E-17"
        izz="0.097658205148634" />
    </inertial>
    <visual>
      <origin
        xyz="0.05321 -0.05009 -0.04838"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/finger0.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/finger0.stl" />
      </geometry>
    </collision>
  </link>
  <joint
    name="finger0_joint"
    type="revolute">
    <origin      
      xyz="0.11278 -0.00482 -0.06085"
      rpy="0 0 0" />
    <parent
      link="gripper_link2" />
    <child
      link="finger0" />
    <axis
      xyz="-0.4811975 -0.22519062  0.84719428" />
    <limit lower="0" upper="0.22" effort="0" velocity="0"/>
  </joint>
  
  <link
    name="finger1">
    <inertial>
      <origin
      xyz="0 0 0"
      rpy="0 0 0" />
      <mass
        value="11.5298327680895" />
      <inertia
        ixx="0.135971148628439"
        ixy="-1.95838647067939E-13"
        ixz="-9.49762849544681E-14"
        iyy="0.097658205148634"
        iyz="2.42861286636753E-17"
        izz="0.097658205148634" />
    </inertial>
    <visual>
      <origin
        xyz="0.07351 -0.00291 -0.01169"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/finger1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/finger1.stl" />
      </geometry>
    </collision>
  </link>
  <joint
    name="finger1_joint"
    type="revolute">
    <origin      
      xyz="0.11278 -0.00482 -0.06085"
      rpy="0 0 0" />
    <parent
      link="gripper_link2" />
    <child
      link="finger1" />
    <axis
      xyz="0.19700433 0.97014849  -0.14142563" />
    <limit lower="0" upper="0.07" effort="0" velocity="0"/>
  </joint>

  <link
    name="finger2">
    <inertial>
      <origin
      xyz="0 0 0"
      rpy="0 0 0" />
      <mass
        value="11.5298327680895" />
      <inertia
        ixx="0.135971148628439"
        ixy="-1.95838647067939E-13"
        ixz="-9.49762849544681E-14"
        iyy="0.097658205148634"
        iyz="2.42861286636753E-17"
        izz="0.097658205148634" />
    </inertial>
    <visual>
      <origin
        xyz="0.05327 0.03648 -0.04625"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/finger2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://rover19_urdf/meshes/finger2.stl"/>
      </geometry>
    </collision>
  </link>
  <joint
    name="finger2_joint"
    type="revolute">
    <origin      
      xyz="0.11278 -0.00482 -0.06085"
      rpy="0 0 0" />
    <parent
      link="gripper_link2" />
    <child
      link="finger2" />
    <axis
      xyz="0.43525535 -0.23528661 -0.8690184" />
    <limit lower="0" upper="0.14" effort="0" velocity="0"/>
  </joint>
</robot>